build/generate_release_index.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/python
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generates jax_releases.html package index.
+
+To update public copy, use:
+gsutil cp jax_releases.html gs://jax-releases/
+"""
+
+import subprocess
+
+FILENAME = "jax_releases.html"
+
+HEADER = """
+<!-- Generated by jax/build/generate_release_index.py, do not update manually! -->
+<html>
+<head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"></head>
+<body>
+"""
+
+FOOTER = "</body>\n</html>\n"
+
+ls_output = subprocess.check_output(["gsutil", "ls", "gs://jax-releases/cuda*"])
```

Comment:
Maybe check this fails gracefully e.g., if your internets is down or you have no permissions?

Reply:
It looks like it prints stderr to the console, so errors from gsutil should be visible. I added some contextual print statements as well.